### PR TITLE
Upgrade module to Django 1.9.

### PR DESCRIPTION
**Pull request by mohannadsh**
CC: @awartani @danaspiegel @waseem-omar

### CODE REVIEW

File: accounts/admin.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 from __future__ import unicode_literals
-
```
Comment (waseem-omar) [inline accounts/admin.py:2]: @mohannadsh Please keep the empty lines, those should separate import groups.

File: accounts/admin.py [hunk 2]
```diff
@@ -17,11 +16,9 @@
 import django.core.urlresolvers
 import django.views.decorators.debug
 from django.utils.translation import ugettext_lazy as _
-
 import signals
 import models
 
-
```
Comment (waseem-omar) [inline accounts/admin.py:24]: @mohannadsh This should be here, there should be two empty lines between imports and next definitions.

File: accounts/admin.py [hunk 3]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
     change_user_password_template = None
     list_display = ('email', 'masquerade', 'first_name', 'last_name',
                     'company', 'is_active', 'is_superuser', 'is_staff',
-                    'get_timezone', 'last_login', )
+                    'get_timezone', 'last_login',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh We used to add a space after the trailing comma

File: accounts/admin.py [hunk 4]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
     fieldsets = (
         (None, {
             'fields': ('email', 'password', 'first_name', 'last_name',
                        ),
         }),
         ('Company', {
-            'fields': ('company', ),
+            'fields': ('company',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 5]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Preferences', {
-            'fields': ('timezone', ),
+            'fields': ('timezone',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 6]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Security', {
             'fields': ('is_active', 'is_superuser', 'is_staff', 'groups',
-                       'user_permissions', ),
+                       'user_permissions',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 7]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Important dates', {
-            'fields': ('last_login', 'created_on', 'updated_on', ),
+            'fields': ('last_login', 'created_on', 'updated_on',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 8]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
     )
     add_fieldsets = (
         (None, {
             'fields': ('email', 'password1', 'password2', 'first_name',
-                       'last_name', ),
+                       'last_name',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 9]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Company', {
-            'fields': ('company', ),
+            'fields': ('company',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 10]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Preferences', {
-            'fields': ('timezone', ),
+            'fields': ('timezone',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 11]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
         }),
         ('Security', {
             'fields': ('is_active', 'is_superuser', 'is_staff', 'groups',
-                       'user_permissions', ),
+                       'user_permissions',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 12]
```diff
@@ -91,48 +90,48 @@ class UserAdmin(django.contrib.auth.admin.UserAdmin):
...
-    readonly_fields = ('last_login', 'created_on', 'updated_on', )
-    actions = ('reset_passwords', )
-    filter_horizontal = ('groups', 'user_permissions', )
-    list_filter = ('is_active', )
+    search_fields = ('email', 'first_name', 'last_name',)
+    ordering = ('email',)
+    readonly_fields = ('last_login', 'created_on', 'updated_on',)
+    actions = ('reset_passwords',)
+    filter_horizontal = ('groups', 'user_permissions',)
+    list_filter = ('is_active',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 13]
```diff
@@ -290,10 +290,10 @@ class LogEntryAdmin(django.contrib.admin.ModelAdmin):
     }
 
     date_hierarchy = 'action_time'
-    list_filter = (ActionFilter, 'content_type', 'user', )
-    search_fields = ('object_repr', 'change_message', )
+    list_filter = (ActionFilter, 'content_type', 'user',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 14]
```diff
@@ -290,10 +290,10 @@ class LogEntryAdmin(django.contrib.admin.ModelAdmin):
...
+    search_fields = ('object_repr', 'change_message',)
     list_display = ('action_time', 'user', 'content_type', 'object_link',
-                    'action', 'change_message', )
+                    'action', 'change_message',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 15]
```diff
@@ -319,24 +319,26 @@ def object_link(self, obj):
...
 
 
 @django.contrib.admin.register(models.Company)
 class CompanyAdmin(django.contrib.admin.ModelAdmin):
-    search_fields = ('name', )
-    list_display = ('name', 'created_on', )
-    readonly_fields = ('created_on', 'updated_on', )
+    search_fields = ('name',)
+    list_display = ('name', 'created_on',)
+    readonly_fields = ('created_on', 'updated_on',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 16]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
 
 @django.contrib.admin.register(models.AuditLogEvent)
 class AuditLogEventAdmin(django.contrib.admin.ModelAdmin):
-    list_filter = ('company', )
-    search_fields = ('user_email', 'message', )
+    list_filter = ('company',)
+    search_fields = ('user_email', 'message',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 17]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
...
     list_display = ('recorded_on', 'user', 'company',
                     'is_masquerading', 'masquerading_user',
-                    'message', )
-    ordering = ('-recorded_on', )
+                    'message',)
+    ordering = ('-recorded_on',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 18]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
...
     list_per_page = 50
     fieldsets = (
         ('', {
-            'fields': ('recorded_on', 'user_id', 'user_email', 'company', ),
+            'fields': ('recorded_on', 'user_id', 'user_email', 'company',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 19]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
...
         }),
         ('Masquerading User', {
-            'fields': ('masquerading_user_email', 'masquerading_user_id', ),
+            'fields': ('masquerading_user_email', 'masquerading_user_id',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 20]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
...
         }),
         ('Audit Log Message', {
-            'fields': ('message', ),
+            'fields': ('message',),
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

File: accounts/admin.py [hunk 21]
```diff
@@ -354,27 +356,27 @@ def save_model(self, request, obj, form, change):
...
         }),
     )
     readonly_fields = ('created_on', 'updated_on', 'recorded_on', 'user_id',
                        'user_email', 'company', 'message',
-                       'masquerading_user_email', 'masquerading_user_id', )
+                       'masquerading_user_email', 'masquerading_user_id',)
```
Comment (waseem-omar) [inline accounts/admin.py]: @mohannadsh Same as above

**Comment by mohannadsh** [comment]
@waseem-omar I'm not sure, what is the standard here? because these changed happened when pycharm did the code format.

**Comment by danaspiegel** [comment]
@mohannadsh @waseem-omar you can adjust your code formatting settings in PyCharm config panel. Standard is spaces after commas, and commas after last elements in collections. See https://www.python.org/dev/peps/pep-0008/

**Comment by awartani** [comment]
@danaspiegel we should add `.editorconfig` as [recommended by Django](https://docs.djangoproject.com/en/1.9/internals/contributing/writing-code/coding-style/). Specially that we're using PyCharm for development.

Checkout [editorconfig.org](editorconfig.org)

cc: @waseem-omar @mohannadsh

File: accounts/models.py [hunk 22]
```diff
@@ -95,7 +97,7 @@ class User(django.contrib.auth.models.AbstractBaseUser,
     email = django.db.models.EmailField(_('Email'), unique=True)
     timezone = timezone_field.TimeZoneField(default='America/New_York')
 
-    company = django.db.models.ForeignKey(Company, null=True, blank=True, related_name='users')
+    company = django.db.models.ForeignKey(Company, null=True, related_name='%(app_label)s_%(class)s_users')
```
Comment (awartani) [inline accounts/models.py:93]: @waseem-omar why `company` should be included in this general purpose library?

File: README.rst [hunk 23]
```diff
@@ -23,22 +23,29 @@ Quick start
...
     url(r'^/', include('accounts.urls')),
 
-3. Run `python manage.py migrate` to create the accounts models.
+4. Run `python manage.py migrate` to create the accounts models.
 
-4. Start the development server admin/ to create users and companies. From Users list view, you can take advantage of the masquerading feature.
+5. Start the development server admin/ to create users and companies. From Users list view, you can take advantage of the masquerading feature.
 
-5. For timezone enablement, add "" to MIDDLEWARE_CLASSES like this::
+6. For timezone enablement, add "" to MIDDLEWARE_CLASSES like this::
```
Comment (waseem-omar) [inline README.rst:44]: @mohannadsh `""` should be `accounts.middleware.TimezoneMiddleware`

File: accounts/models.py (hunk 22 above)
Comment (waseem-omar) [inline accounts/models.py:93]: @awartani It is included but have `null=True`, so whoever don't want to use it they can just ignore it and remove it from Admin site. However, I think `company` is a general purpose field.

File: accounts/models.py [hunk 24]
```diff
@@ -137,20 +139,25 @@ def email_user(self, subject, message, from_email=None, **kwargs):
...
+class AbstractAuditLogEventBase(django.db.models.Model):
     created_on = django.db.models.DateTimeField(auto_now_add=True)
     updated_on = django.db.models.DateTimeField(auto_now=True)
     recorded_on = django.db.models.DateTimeField(auto_now_add=True)
 
     user_id = django.db.models.IntegerField(_('User ID'), db_index=True)
     user_email = django.db.models.EmailField(_('User Email'), db_index=True)
-    company = django.db.models.ForeignKey('accounts.Company')
+    company = django.db.models.ForeignKey('accounts.Company', null=True, on_delete=django.db.models.SET_NULL,
+                                          related_name='%(app_label)s_%(class)s_users')
```
Comment (waseem-omar) [inline accounts/models.py:142]: @mohannadsh related_name should reference to the current model, I think it should be something like `%(app_label)s_%(class)s_logs`

**Comment by danaspiegel** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;DWI0oXd8&#x2F;9-1-upgrade-django-users-plus-to-python-1-9-week1"> (1) Upgrade `django-users-plus` to Python 1.9 [Week1]</a>

**Pull request merged**